### PR TITLE
feat: support PLONK-in-Wizard for smallfields

### PR DESCRIPTION
**Pull request by ivokub**
This PR fixes PLONK-in-Wizard for smallfields. It also refactors the PLONK circuit interfaces to have a unified implementation across different variations.

~Do not merge yet - need gnark dependency to be merged and will merge https://github.com/Consensys/linea-monorepo/commit/e4ba2796a6b699058280292481be85dc2d35106f~

### Checklist

* [x] I wrote new tests for my new core changes.
* [x] I have successfully ran tests, style checker and build against my new changes locally.
* [x] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> Implements a PLONK builder with wide commitments, wires MiMC/range-check integrations, and updates PLONK-in-Wizard to handle small-field workflows (including HCP decomposition), with tests and deps bumped.
> 
> - **Core (PLONK-in-Wizard)**:
>   - Handle commitment coin (`HCP`) as extension field by decomposing into base-field columns `HCP_BASE_[0..3]` and enforcing correctness via global constraints.
>   - Adjust gate expression to use committed output selector and HCP decomposition; assign `HcpEl` in prover.
> - **Builder / API**:
>   - Introduce `plonkinternal/plonkbuilder` providing `Builder` that wraps gnark SCS and implements `frontend.WideCommitter` with placeholder hint; exposes wire-constraint/KV/PLONK APIs.
>   - Switch compilation to `frontend.CompileU32` with `plonkbuilder.From(scs.NewBuilder[constraint.U32])`.
> - **External integrations**:
>   - Migrate external range-checker and MiMC hasher builders to use `plonkbuilder.Builder` and new wire-constraint APIs.
>   - Remove old fake wide-committer (`protocol/query/gnark_widecommiter.go`).
> - **Refactors**:
>   - Move imports from `protocol/internal/plonkinternal` to `protocol/plonkinternal` across compilers and tests; delete `prover/crypto/plonkcircuitregistry` (unused).
> - **Tests**:
>   - Update witnesses to small-field `AsWitnessPublic([]any)`; add tests for wide commit + range-check; assert non-zero wide commit outputs.
> - **Dependencies**:
>   - Bump `gnark` and `gnark-crypto` versions in `go.mod`/`go.sum`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f12e875ca9a9bcc342da5ad40495d30f3b481fce. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: prover/utils/gnarkutil/gnarkutil.go [hunk 1]
```diff
@@ -22,46 +23,29 @@ func AllocateSliceExt(n int) []gnarkfext.Element {
...
+		for _, w := range v {
+			witChan <- w
+		}
+		close(witChan)
+	}()
 
 	if err := wit.Fill(len(v), 0, witChan); err != nil {
 		panic(err)
 	}
+	wg.Wait()
```
[the remark this answers is not in the document]
Comment (ivokub) [reply prover/utils/gnarkutil/gnarkutil.go]: Removed the waitgroup, indeed it is not needed.
